### PR TITLE
QML: Mark required properties as such

### PR DESCRIPTION
**Pull request by Holzhaus**
Before, this was postponed because it only works with Qt 5.15+. In #4649 we dropped support for Qt 5.x anyway, so it makes sense to add it.

### CODE REVIEW

File: res/qml/ComboBox.qml [hunk 1]
```diff
@@ -12,9 +12,11 @@ ComboBox {
     delegate: ItemDelegate {
         id: itemDlgt
 
+        required property int index
+
         width: parent.width
-        highlighted: root.highlightedIndex === index
-        text: root.textAt(index)
+        highlighted: root.highlightedIndex === this.index
```
Comment (uklotzde) [inline res/qml/ComboBox.qml:18]: Does this fix the various *property does not exist* warnings I saw when starting the Qt6 QML version?

**Review by uklotzde — APPROVED** [review]

File: res/qml/ComboBox.qml (hunk 1 above)
Comment (Holzhaus) [reply res/qml/ComboBox.qml:18]: If you mean the "verticalCenter" warnings, then no, these are adifferent issue.

But this fixes a bunch of qmllint warnings where it complains about unqualified lookups.

https://doc.qt.io/qt-6/qml-codingconventions.html#required-properties

**Comment by Swiftb0y** [comment]
I'll take a look tomorrow. I need to read up on the details or `required` on delegates.

**Comment by uklotzde** [comment]
As long as the CI builds succeeds merging QML PRs is acceptable.

@Swiftb0y Please press merge if you don't spot any obvious flaws.

**Review by Swiftb0y — APPROVED** [review]
LGTM. Thanks.
Sorry for not taking a look yesterday.

**Pull request merged**